applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c

```diff
@@ -4,3 +4,173 @@
+
+/* When BROADCAST_ENQUEUE_COUNT > 1 we can enqueue enough buffers to ensure that
+ * the controller is never idle
+ */
+#define BROADCAST_ENQUEUE_COUNT 2U
+#define TOTAL_BUF_NEEDED (BROADCAST_ENQUEUE_COUNT * CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT)
+
+BUILD_ASSERT(CONFIG_BT_ISO_TX_BUF_COUNT >= TOTAL_BUF_NEEDED,
+	     "CONFIG_BT_ISO_TX_BUF_COUNT should be at least "
+	     "BROADCAST_ENQUEUE_COUNT * CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT");
+
+static struct bt_audio_lc3_preset lc3_preset = BT_AUDIO_LC3_BROADCAST_PRESET_48_4_1;
+// TODO: Currently only 1 stream is supported
+static struct bt_audio_stream streams[CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT];
+static struct bt_audio_broadcast_source *broadcast_source;
+
+NET_BUF_POOL_FIXED_DEFINE(tx_pool, TOTAL_BUF_NEEDED, BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU), 8,
+			  NULL);
+
+static uint32_t send_req_cnt;
+static uint32_t sent_cnt;
+static bool stopping;
+
+static void stream_sent_cb(struct bt_audio_stream *stream)
+{
+	int32_t diff;
+
+	sent_cnt++;
+
+	diff = send_req_cnt - sent_cnt;
+
+	// TODO: Use the atomic functionality from ble_trans
+	// if (diff < 0) {
+	// 	/* Should not happen */
+	// 	LOG_WRN("Sending underrun");
+	// } else if (diff > 0) {
+	// 	LOG_WRN("Sending overrun");
+	// }
+
+	if ((sent_cnt % 1000U) == 0U) {
+		LOG_INF("Sent %u total ISO packets", sent_cnt);
+	}
+}
+
+static void stream_started_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_STREAMING;
+	
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+
+	LOG_INF("Broadcast source started");
+}
+
+static void stream_stopped_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Broadcast source stopped");
+}
+
+static struct bt_audio_stream_ops stream_ops = { .sent = stream_sent_cb,
+						 .started = stream_started_cb,
+						 .stopped = stream_stopped_cb };
+
+int le_audio_config_get(void)
+{
+	return 0;
+}
+
+int le_audio_volume_up(void)
+{
+	return 0;
+}
+
+int le_audio_volume_down(void)
+{
+	return 0;
+}
+
+int le_audio_volume_mute(void)
+{
+	return 0;
+}
+
+int le_audio_play(void)
+{
+	return 0;
+}
+
+int le_audio_pause(void)
+{
+	return 0;
+}
+
+int le_audio_send(uint8_t const *const data, size_t size)
+{
+	// TODO: Implement the synchronization workaround created by Audun here? (OR ONLY FOR CIS?)
+	// It this upstreamed now?
+
+	int ret;
+	struct net_buf *buf;
+
+	if (stopping) {
+		return -ECANCELED;
+	}
+
+	buf = net_buf_alloc(&tx_pool, K_FOREVER);
+	if (buf == NULL) {
+		LOG_WRN("Could not allocate buffer when sending");
+		return -ENOMEM;
+	}
+
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+	net_buf_add_mem(buf, data, size);
+
+	ret = bt_audio_stream_send(streams, buf);
+	if (ret < 0) {
+		LOG_WRN("Unable to broadcast data: %d", ret);
+		// TODO: Must unref when finished sending?
+		net_buf_unref(buf);
+		return ret;
+	}
+
+	send_req_cnt++;
```

Comment:
More statistics that we probably don't need.

Reply:
Same as above:
The thought here is to implement the same functionality as 'atomic_t iso_tx_pool_alloc' in ble_trans.c to keep track of overruns/underruns.
I guess in practice catching if le_audio_send() is called faster than packages are being sent